pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/groupby/NoDictionarySingleColumnGroupKeyGenerator.java

```diff
@@ -164,8 +166,113 @@ private Map createGroupKeyMap(DataType keyType) {
 
   @Override
   public void generateKeysForBlock(TransformBlock transformBlock, int[][] groupKeys) {
```

Comment:
Shouldn't we make the same enhancement to `NoDictionaryMultiColumnGroupKeyGenerator` as well ?

Reply:
This function is already implemented for `NoDictionaryMultiColumnGroupKeyGenerator`